### ISSUE TITLE
Promotion rule expiration is dependent on system time

### ISSUE THREAD

**Issue by nickdelnano** [openark/orchestrator#585]
Orchestrator [uses the system time when setting the LastUpdated field of a new CandidateDatabaseInstance](https://github.com/github/orchestrator/blob/master/go/inst/candidate_database_instance.go#L37). 
This stores the system time with an offset to UTC: (I've set the system time to PDT)
```
/usr/local/orchestrator # sqlite3 orchestrator.sqlite3 "select * from candidate_database_instance;"
8fa4a00f0156|3306|2018-08-20 20:09:12.395052589-07:00|1|prefer
```

When [ExpireCandidateInstances](https://github.com/github/orchestrator/blob/master/go/inst/instance_dao.go#L2572) looks to expire entries, it uses the query:
```
delete from candidate_database_instance where last_suggested < NOW() - INTERVAL ? MINUTE
```
where the NOW() function is converted to datetime('now") for sqlite, which implicitly uses UTC. Sqlite does not parse the offset listed in the last_suggested column, only the text in the format "YYYY-MM-DD HH:MM:SS.SSS" (see section 2.2 of sqlite types manual https://www.sqlite.org/datatype3.html).

This causes promotion rules to be forgotten in near instant time on a machine whose time is set 1 hour or more behind UTC. I believe this also effects MySQL users the same.

I've reproduced this with the project's docker setup; the reproduced output and the steps to reproduce are listed in https://github.com/github/orchestrator/pull/584.

**Comment by shlomi-noach** [comment]
Thank you for this analysis. Before looking into it, do you happen to have a suggestion for remediation?

**Comment by shlomi-noach** [comment]
Closed by #591
